### DISCUSSION TITLE
Help with Sonoff 4CH pro3

### DISCUSSION THREAD

**Discussion by mwouda** [RavenSystem/esp-homekit-devices#2261] [MEPLHAA Scripts]
Hi, 

I need some help. I've several sonoff 4ch pro's running for years without problems. Bu no i updated to latest version and i tryed to get the same config in the new format working but i does not work. 

This is what is connected: 
1 = gate-led (lamp)
2 = Gate-spot (lamp)
3 = Pipe defrost (heater) basicly a switch
4 = Gate open\close

This is what i now have that is accepted but bot working. 

`{"c":
{"io":[[[4,5,12,15,13],2],[[0,9,10,14],6]],"l":13},
"o":0,
"tz":"CET-1CEST-2,M3.5.0/2,M10.5.0/3",
"b":[{"g":0,"t":5}],
"n":"Sonoff-meterkast",
"a":[{
	"0":{"r":[{"g":12,"v":0}]},
	"1":{"r":[{"g":12,"v":1}]},
	"t":1,"s":0,"b":[{"g":0,"t":1}]
	},{
	"0":{"r":[{"g":5,"v":0}]},
	"1":{"r":[{"g":5,"v":1}]},
	"t":1,"s":0,"b":[{"g":9,"t":1}]
	},{
	"0":{"r":[{"g":4,"v":0}]},
	"1":{"r":[{"g":4,"v":1}]},
	"t":1,"s":0,"b":[{"g":10,"t":1}]
	},{
	"t":40,"s":1,"d":23,"c":25,
	"0":{"r":[{"g":15,"v":1,"i":0.5}]},"1":{"a":0},
	"1":{"a":0},
	"2":{"r":[{"g":15,"i":0.5},{"g":15,"v":1,"i":0.5},{"g":15,"v":1,"i":0.5}]},
	"3":{"a":2},
	"10":{"a":0},"es":[{"i":1,"1":{"m":[[-1,2]]}}]}]}`

After reboot and adding the device to homekit i keep getting " not responding"  while the device is online in the network
please any help and hint on what i'm doing wrong is appiciated.

**Comment by mwouda** [comment]
Oh and also when i press any buttons on the app or on the device.. nothing happens

**Comment by RavenSystem** [comment]
Try enabling internal pullup resistors for inputs:
[[0,9,10,14],6,1]

**Comment by mwouda** [comment]
Oke.. I tryed that. 
The buttons on the device 1-3 work. the gate nr. 4 does nothing. 
Also in het homekit i still see "no reaction" and none of the buttons work.

any ideas?

**Reply by mwouda** [reply]
Oh and i have the HAA manager.. That also does not work. When i try to switch to setup or if i what it to reboot.

**Reply by RavenSystem** [reply]
You need to declare any use with the GPIO 14. Currently, it is not used anywhere.

Be sure that HAA Home manager App is up to date.

**Reply by mwouda** [reply]
HAA manager is V 3.7.4. (it automaticly updates)

GPIO 14 is configured declared internal pullup.. This is the latest code: 

`{"c":
{"io":[[[4,5,12,15,13],2],[[0,9,10,14],6,1l":13},
"o":0,
"tz":"CET-1CEST-2,M3.5.0/2,M10.5.0/3",
"b":[{"g":0,"t":5}],
"n":"Sonoff-meterkast",
"a":[{
	"0":{"r":[{"g":12,"v":0}]},
	"1":{"r":[{"g":12,"v":1}]},
	"t":1,"s":0,"b":[{"g":0,"t":1}]
	},{
	"0":{"r":[{"g":5,"v":0}]},
	"1":{"r":[{"g":5,"v":1}]},
	"t":1,"s":0,"b":[{"g":9,"t":1}]
	},{
	"0":{"r":[{"g":4,"v":0}]},
	"1":{"r":[{"g":4,"v":1}]},
	"t":1,"s":0,"b":[{"g":10,"t":1}]
	},{
	"t":40,"s":1,"d":23,"c":25,
	"0":{"r":[{"g":15,"v":1,"i":0.5}]},"1":{"a":0},
	"1":{"a":0},
	"2":{"r":[{"g":15,"i":0.5},{"g":15,"v":1,"i":0.5},{"g":15,"v":1,"i":0.5}]},
	"3":{"a":2},
"10":{"a":0},"es":[{"i":1,"1":{"m":[[-1,2]]}}]}]}`
